datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -946,13 +946,29 @@ pub(crate) fn parquet_column<'a>(
 ) -> Option<(usize, &'a FieldRef)> {
     let (root_idx, field) = arrow_schema.fields.find(name)?;
     if field.data_type().is_nested() {
-        // Nested fields are not supported and require non-trivial logic
-        // to correctly walk the parquet schema accounting for the
-        // logical type rules - <https://github.com/apache/parquet-format/blob/master/LogicalTypes.md>
-        //
-        // For example a ListArray could correspond to anything from 1 to 3 levels
-        // in the parquet schema
-        return None;
+        match field.data_type() {
```

Comment:
Perhaps we should extract the repeated code into a helper function? Could also reduce nesting with an early return for the case where it's not nested. So something like:

```suggestion
pub(crate) fn parquet_column<'a>(
    parquet_schema: &SchemaDescriptor,
    arrow_schema: &'a Schema,
    name: &str,
) -> Option<(usize, &'a FieldRef)> {
    let (root_idx, field) = arrow_schema.fields.find(name)?;
    if !field.data_type().is_nested() {
        let parquet_idx = find_parquet_idx(parquet_schema, root_idx)?;
        return Some((parquet_idx, field));
    }
    // Nested field
    match field.data_type() {
        DataType::Struct(_) => {
            let parquet_idx = find_parquet_idx(parquet_schema, root_idx)?;
            Some((parquet_idx, field))
        }
        _ => {
            if field.data_type().is_nested() {
                // Nested fields are not supported and require non-trivial logic
                // to correctly walk the parquet schema accounting for the
                // logical type rules - <https://github.com/apache/parquet-format/blob/master/LogicalTypes.md>
                //
                // For example a ListArray could correspond to anything from 1 to 3 levels
                // in the parquet schema
                None
            } else {
                let parquet_idx = find_parquet_idx(parquet_schema, root_idx)?;
                Some((parquet_idx, field))
            }
        }
    }
}

// Helper function to find parquet index - TBD: this could be more efficient
fn find_parquet_idx(parquet_schema: &SchemaDescriptor, root_idx: usize) -> Option<usize> {
    (0..parquet_schema.columns().len())
        .find(|x| parquet_schema.get_column_root_idx(*x) == root_idx)
}
```

Reply:
Sure, I can do it